packages/@uppy/companion/src/server/Uploader.js

```diff
@@ -492,18 +502,25 @@ class Uploader {
     })
   }
 
-  /**
-   * Upload the file to S3 while it is still being downloaded.
-   */
-  uploadS3Multipart () {
+  beginTailReadStream () {
     const file = createTailReadStream(this.path, {
```

Comment:
I'm using the tail read stream in this case to monitor the progress of the file being received so the individual upload parts can begin being hashed as soon as they're received.

Unless your plan is to move away from async download/upload entirely? If that is the case, a lot of this could be simplified I suppose.

Reply:
yes from the note in [this PR](https://github.com/transloadit/uppy/pull/2114), the intention is to move away from parallel upload/download for now, since the usage of the library has seemed [problematic](https://github.com/transloadit/uppy/issues/1457) so far